experimental/rsocket-src/RSocketServer.cpp

```diff
@@ -0,0 +1,113 @@
+
+#include "rsocket/RSocketServer.h"
+#include <folly/ExceptionWrapper.h>
+#include <folly/io/async/EventBaseManager.h>
+#include "rsocket/RSocketErrors.h"
+#include "src/FrameTransport.h"
+
+namespace rsocket {
+
+class RSocketConnection : public reactivesocket::ServerConnectionAcceptor {
+ public:
+  RSocketConnection(OnSetupNewSocket onSetup) : onSetup_(std::move(onSetup)) {}
+  ~RSocketConnection() {
+    LOG(INFO) << "RSocketServer => destroy the connection acceptor";
+  }
+
+  void setupNewSocket(
+      std::shared_ptr<FrameTransport> frameTransport,
+      ConnectionSetupPayload setupPayload,
+      Executor& executor) {
+    onSetup_(std::move(frameTransport), std::move(setupPayload), executor);
+  }
+  void resumeSocket(
+      std::shared_ptr<FrameTransport> frameTransport,
+      ResumeIdentificationToken,
+      ResumePosition,
+      Executor& executor) {
+    //      onSetup_(std::move(frameTransport), std::move(setupPayload));
+  }
+
+ private:
+  OnSetupNewSocket onSetup_;
+};
+
+RSocketServer::RSocketServer(
+    std::unique_ptr<ConnectionAcceptor> connectionAcceptor)
+    : lazyAcceptor(std::move(connectionAcceptor)) {}
+
+void RSocketServer::start(OnAccept onAccept) {
+  if (!acceptor_) {
+    LOG(INFO) << "RSocketServer => initialize connection acceptor on start";
+    acceptor_ = std::make_unique<RSocketConnection>([
+      this,
+      onAccept = std::move(onAccept)
+    ](std::shared_ptr<FrameTransport> frameTransport,
+      ConnectionSetupPayload setupPayload,
+      Executor & executor_) {
+      LOG(INFO) << "RSocketServer => received new setup payload";
+
+      std::shared_ptr<RequestHandler> requestHandler;
+      try {
+        requestHandler = onAccept(
+            std::make_unique<ConnectionSetupRequest>(std::move(setupPayload)));
+      } catch (RSocketError& e) {
+        // TODO emit ERROR ... but how do I do that here?
+        frameTransport->close(
+            folly::exception_wrapper{std::current_exception(), e});
+        return;
+      }
+      LOG(INFO) << "RSocketServer => received request handler";
+
+      auto rs = StandardReactiveSocket::disconnectedServer(
+          // we know this callback is on a specific EventBase
+          executor_,
+          std::move(requestHandler),
+          Stats::noop());
```

Comment:
also we need to figure out how to pass in the right Stats instance. 
Maybe a variant of RSocketServer::start(OnAccept onAccept, ServerSetup setupParams) ?